packages/page-accounts/src/modals/Derive.tsx

```diff
@@ -85,16 +72,15 @@ function Derive ({ className = '', from, onClose }: Props): React.ReactElement {
   const { queueAction } = useContext(StatusContext);
   const [source] = useState(() => keyring.getPair(from));
   const [isBusy, setIsBusy] = useState(false);
+  const [{ isNameValid, name }, setName] = useState({ isNameValid: false, name: '' });
+  const [{ isPassValid, password }, setPassword] = useState({ isPassValid: false, password: '' });
   const [{ address, deriveError }, setDerive] = useState<DeriveAddress>({ address: null, deriveError: null });
   const [isConfirmationOpen, toggleConfirmation] = useToggle();
   const [{ isLocked, lockedError }, setIsLocked] = useState<LockState>({ isLocked: source.isLocked, lockedError: null });
-  const [{ isNameValid, name }, setName] = useState({ isNameValid: false, name: '' });
-  const [{ isPassValid, password }, setPassword] = useState({ isPassValid: false, password: '' });
-  const [{ isPass2Valid, password2 }, setPassword2] = useState({ isPass2Valid: false, password2: '' });
   const [{ isRootValid, rootPass }, setRootPass] = useState({ isRootValid: false, rootPass: '' });
   const [suri, setSuri] = useState('');
   const debouncedSuri = useDebounce(suri);
-  const isValid = !!address && !deriveError && isNameValid && isPassValid && isPass2Valid;
```

Comment:
why a logical change? not used?

Reply:
Actually the logic is not changed, but this not obvious because the same name is used for the variable that works differently now. `isPassValid` is handled by `CreateAccountInputs` is such a way that it is equivalent to the older `isPassValid && isPass2Valid` expression. The second password is handled internally by the `CreateAccountInputs` component, just as it was handled in the Create modal. Do you agree or did I miss something?

Reply:
By "logic is not changed" I meant that functionally this works the same as before. And yes, the second password is not used here as it is handled internally by the `CreateAccountInputs`

Reply:
got it, thx